lib/ansible/modules/web_infrastructure/pmp.py

```diff
@@ -0,0 +1,296 @@
+                return acc
+        raise RuntimeError('ACCOUNT not found!')
+
+    def getAccountPwd(self):
+        url = self.format(
+            'https://{HOST}:{PORT}/restapi/json/v1/resources/{resourceId}/accounts/{accountId}/password?AUTHTOKEN={TOKEN}')
+        res = json.loads(open_url(url, validate_certs=self.validate_certs, use_proxy=self.use_proxy).read())['operation']
+        status = res['result']['status']
+        if status == 'Failed':
+            raise RuntimeError(res['result']['message'])
+        return res['Details']['PASSWORD']
+
+    def getResourceByName(self, name):
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources?AUTHTOKEN={TOKEN}')
+        resources = json.loads(open_url(url, validate_certs=self.validate_certs, use_proxy=self.use_proxy).read())
+        for res in resources['operation']['Details']:
+            if res['RESOURCE NAME'] == name:
+                return res
+        raise RuntimeError('RESOURCE not found!')
+
+    def getPassword(self, server, username):
+        resource = self.getResourceByName(server)
+        self.data['resourceId'] = resource['RESOURCE ID']
+        account = self.getAccountResourceByName(username)
+        self.data['accountId'] = account['ACCOUNT ID']
+        return self.getAccountPwd()
+
+    def createResource(self, resourcename, accountname, password, ownername, resource_type='Linux', group=''):
+        data = {
+            "operation": {
+                "Details": {
+                    "RESOURCENAME": resourcename,
+                    "ACCOUNTNAME": accountname,
+                    "RESOURCETYPE": resource_type,
+                    "PASSWORD": password,
+                    "OWNERNAME": ownername,
+                }
+            }
+        }
+        if group:
+            data["operation"]["Details"]["RESOURCEGROUPNAME"] = group
+        headers = {
+            'Content-Type': 'text/json',
+        }  # admin test
+
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources?AUTHTOKEN={TOKEN}')
+        data = json.dumps(data, sort_keys=False)
+        params = urllib.urlencode({'INPUT_DATA': data})
+        r = open_url(url, method="POST", headers=headers, data=params, validate_certs=self.validate_certs, use_proxy=self.use_proxy)
+        res = json.loads(r.read())['operation']['result']
+        status = res['status']
+        if status == 'Failed':
+            raise RuntimeError(res['message'])
+        return status
+
+
+def get_password(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    password = pmp.getPassword(params['resource_name'], params['account_name'])
+    return {'status': 'OK', 'password': password}
+
+
+def create_host(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    status = pmp.createResource(params['resource_name'], params['account_name'], params['password'], params['owner'],
+                                group=params['group'])
+    return {'status': status}
```

Comment:
Since the very first thing that both of the above do is create a ```PasswordManagerPro``` object and then they just lightly wrap a method call on that object, it seems like they should really be integrated into the methods they're calling.